app/services/create_post.rb

```diff
@@ -1,6 +1,6 @@
 class CreatePost
   def self.call(group, membership, post_type_id)
-    if post_type_id == PostType::LEADER_POST_ID
+    if (post_type_id == PostType::LEADER_POST_ID && group.leader.present?)
```

Comment:
I recommend to enable `rubocop` or check it sometimes, it will help to keep the consistency in the codebase.
https://rubocop.readthedocs.io/en/latest/cops_style/#styleparenthesesaroundcondition